jormungandr/src/fragment/expirations.rs

```diff
@@ -0,0 +1,93 @@
+use slab::Slab;
+use std::{
+    collections::{BTreeMap, HashSet},
+    time::{Duration, Instant},
+};
+
+/// A synchronous alternative to tokio's `DelayQueue` intended to be
+/// periodically polled for expired entries.
+pub struct Expirations<T> {
+    slab: Slab<ExpirationEntry<T>>,
+    sorted: BTreeMap<Instant, HashSet<usize>>,
```

Comment:
Use the `Key` to remain consistent

```suggestion
    sorted: BTreeMap<Instant, HashSet<Key>>,
```